osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -90,8 +90,14 @@ public abstract class SongSelect : OsuScreen, IKeyBindingHandler<GlobalAction>
         [Cached(Type = typeof(IBindable<IReadOnlyList<Mod>>))]
         private readonly Bindable<IReadOnlyList<Mod>> mods = new Bindable<IReadOnlyList<Mod>>(Array.Empty<Mod>()); // Bound to the game's mods, but is not reset on exiting
 
-        protected SongSelect()
+        protected virtual void ApplyFilterToCarousel(FilterCriteria criteria) => Carousel.Filter(criteria);
+
+        [BackgroundDependencyLoader(true)]
+        private void load(BeatmapManager beatmaps, AudioManager audio, DialogOverlay dialog, OsuColour colours, SkinManager skins, ScoreManager scores)
         {
+            // initial value transfer is requried for FilterControl (it uses our re-cached bindables in its asynd loac for the initial filter).
```

Comment:
Sorry for doing this, but
```suggestion
            // initial value transfer is required for FilterControl (it uses our re-cached bindables in its async load for the initial filter).
```